### PR TITLE
feat: reading packages catalog from yaml file

### PR DESCRIPTION
**Pull request by leoporoli**

### CODE REVIEW

**Comment by Dartoxian** [comment]
@leoporoli apologies I was looking at the wrong pr for the automerge changes above

File: server/catalog/package_data_from_catalog.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		return nil, stacktrace.Propagate(err, "an error occurred parsing the URL with scheme '%s' for package '%s'", packageURLPrefixedWithHttps, packageName)
+	}
+	if strings.ToLower(parsedURL.Host) != githubDomain {
+		return nil, stacktrace.NewError("an error occurred parsing the URL of package. We only support packages on Github for now but got '%v'", packageName)
+	}
+
+	splitPath := cleanPathAndSplit(parsedURL.Path)
+
+	repositoryOwner := splitPath[0]
+	repositoryName := splitPath[1]
```
Comment (adschwartz) [inline server/catalog/package_data_from_catalog.go:67]: is it possible this could yield a NIL panic?

**Review by adschwartz — APPROVED** [review]
Approved but have a look at a possible NIL issue

File: server/catalog/package_data_from_catalog.go (hunk 1 above)
Comment (leoporoli) [reply server/catalog/package_data_from_catalog.go:67]: yes, if for some reason one row in the catalog is wrong that could happen, good catch, let me add a checker

**Pull request merged**